### PR TITLE
treewide (darwin): fix or enable darwin build for many packages (6)

### PR DESCRIPTION
**Pull request by siraben**
###### Motivation for this change
See #114976 for details.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 116600` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>13 packages built:</summary>
  <ul>
    <li>apparix</li>
    <li>bbe</li>
    <li>bcunit</li>
    <li>catcli</li>
    <li>cdi2iso</li>
    <li>cpcfs</li>
    <li>dejsonlz4</li>
    <li>epstool</li>
    <li>git-fire</li>
    <li>idsk</li>
    <li>ministat</li>
    <li>mpd-mpris</li>
    <li>zdelta</li>
  </ul>
</details>

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 116600` at 2d0392f5 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built successfully:</summary>
  <ul>
    <li><a href="https://gist.github.com/2a002f25a50a3da27be45dacf1fbddf8">cdi2iso</a></li></li>
    <li><a href="https://gist.github.com/edcd5b0fd60559dbb6111a51eca00b6b">dejsonlz4</a></li></li>
    <li><a href="https://gist.github.com/1636889071f7ff43fccba2f70dad5b86">epstool</a></li></li>
    <li><a href="https://gist.github.com/85cdc394264d38fb50600c2e0509920c">zdelta</a></li></li>
  </ul>
</details>
<details>
  <summary>9 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/tools/compression/zdelta/default.nix:19:3:
```
   |
19 |   meta = with lib; {
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/tools/graphics/epstool/default.nix:22:15:
```
   |
22 |   patches = [ ./gcc43.patch ];
   |               ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`buildPhase` should probably contain `runHook preBuild` and `runHook postBuild`.

Near pkgs/tools/compression/dejsonlz4/default.nix:12:5:
```
   |
12 |     buildPhase = ''
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/tools/cd-dvd/cdi2iso/default.nix:16:3:
```
   |
16 |   installPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/tools/compression/dejsonlz4/default.nix:16:5:
```
   |
16 |     installPhase = ''
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/tools/compression/zdelta/default.nix:14:3:
```
   |
14 |   installPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/tools/graphics/epstool/default.nix:18:3:
```
   |
18 |   installPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl2` is a deprecated license, check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/tools/cd-dvd/cdi2iso/default.nix:24:5:
```
   |
24 |     license = licenses.gpl2;
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl2` is a deprecated license, check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/tools/graphics/epstool/default.nix:27:5:
```
   |
27 |     license = licenses.gpl2;
   |     ^
```
</li>
  </ul>
</details>

File: pkgs/tools/compression/dejsonlz4/default.nix [hunk 1]
```diff
@@ -10,7 +10,7 @@ stdenv.mkDerivation rec {
     };
 
     buildPhase = ''
-      gcc -Wall -o dejsonlz4 src/dejsonlz4.c src/lz4.c
+      ${stdenv.cc.targetPrefix}cc -o dejsonlz4 src/dejsonlz4.c src/lz4.c
```
Comment (knedlsepp) [inline pkgs/tools/compression/dejsonlz4/default.nix:13]: I'm curious what the difference to `$CC` is in this case.

File: pkgs/tools/cd-dvd/cdi2iso/default.nix [hunk 2]
```diff
@@ -9,16 +9,20 @@ stdenv.mkDerivation rec {
     sha256 = "0fj2fxhpr26z649m0ph71378c41ljflpyk89g87x8r1mc4rbq3kh";
   };
 
+  postPatch = ''
+    substituteInPlace Makefile --replace "gcc" "${stdenv.cc.targetPrefix}cc"
```
Comment (knedlsepp) [inline pkgs/tools/cd-dvd/cdi2iso/default.nix:13]: $CC?

File: pkgs/tools/compression/zdelta/default.nix [hunk 3]
```diff
@@ -1,16 +1,24 @@
...
 
 stdenv.mkDerivation rec {
-  name = "zdelta-2.1";
-  builder = ./builder.sh;
+  pname = "zdelta";
+  version = "2.1";
+
   src = fetchurl {
-    url = "${meta.homepage}/downloads/${name}.tar.gz";
+    url = "http://cis.poly.edu/zdelta/downloads/${pname}-${version}.tar.gz";
```
Comment (SuperSandro2000) [inline pkgs/tools/compression/zdelta/default.nix:8]: Redirects to https://cse.engineering.nyu.edu/zdelta/downloads/zdelta-2.1.tar.gz and then 404s.

File: pkgs/tools/compression/dejsonlz4/default.nix (hunk 1 above)
Comment (siraben) [reply pkgs/tools/compression/dejsonlz4/default.nix:13]: Personal preference. $CC would work as well but I like to be more explicit about cross compilation prefixes.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 116600` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
13 packages built:
</summary>
<ul>
<li>apparix</li>
<li>bbe</li>
<li>bcunit</li>
<li>catcli</li>
<li>cdi2iso</li>
<li>cpcfs</li>
<li>dejsonlz4</li>
<li>epstool</li>
<li>git-fire</li>
<li>idsk</li>
<li>ministat</li>
<li>mpd-mpris</li>
<li>zdelta</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
apparix:
<details>


warning: unclear-gpl
`gpl2` is a deprecated license, check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/tools/misc/apparix/default.nix:17:5:

```
   |
17 |     license = licenses.gpl2;
   |     ^
```

See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md
</details>
</details>

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 116600` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>21 packages built:</summary>
  <ul>
    <li>altermime</li>
    <li>apparix</li>
    <li>bash_unit</li>
    <li>bbe</li>
    <li>bcunit</li>
    <li>catcli</li>
    <li>cdi2iso</li>
    <li>cpcfs</li>
    <li>dejsonlz4</li>
    <li>dhcping</li>
    <li>epstool</li>
    <li>git-fire</li>
    <li>httperf</li>
    <li>idsk</li>
    <li>imgurbash2</li>
    <li>ministat</li>
    <li>mpd-mpris</li>
    <li>sfeed</li>
    <li>spirv-cross</li>
    <li>vimer</li>
    <li>zdelta</li>
  </ul>
</details>

File: pkgs/tools/compression/zdelta/default.nix (hunk 3 above)
Comment (siraben) [reply pkgs/tools/compression/zdelta/default.nix:8]: Maybe you have automatic HTTPS redirection on. It curls fine.

Comment (siraben) [reply pkgs/tools/compression/zdelta/default.nix]: Using https://web.archive.org/web/20160316212948/http://cis.poly.edu/zdelta/ as the homepage URL.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 116600` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
5 packages built:
</summary>
<ul>
<li>altermime</li>
<li>cdi2iso</li>
<li>dejsonlz4</li>
<li>epstool</li>
<li>zdelta</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 116600` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
21 packages built:
</summary>
<ul>
<li>altermime</li>
<li>apparix</li>
<li>bash_unit</li>
<li>bbe</li>
<li>bcunit</li>
<li>catcli</li>
<li>cdi2iso</li>
<li>cpcfs</li>
<li>dejsonlz4</li>
<li>dhcping</li>
<li>epstool</li>
<li>git-fire</li>
<li>httperf</li>
<li>idsk</li>
<li>imgurbash2</li>
<li>ministat</li>
<li>mpd-mpris</li>
<li>sfeed</li>
<li>spirv-cross</li>
<li>vimer</li>
<li>zdelta</li>
</ul>
</details>

**Pull request merged**